Core/include/Acts/Material/MaterialInteraction.hpp

```diff
@@ -9,13 +9,52 @@
 #pragma once
 
 #include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Detector/DetectorVolume.hpp"
 #include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/TrackingVolume.hpp"
 #include "Acts/Material/MaterialSlab.hpp"
 
 namespace Acts {
 
 class Surface;
-class TrackingVolume;
+
+/// @brief The Material interaction volume struct
+/// It acts as a switch between detctor and tracking volume
+/// as long as those co-exist alongside
+struct InteractionVolume {
+  /// The tracking volume
+  const TrackingVolume* trackingVolume = nullptr;
+  /// The detector volume
+  const Experimental::DetectorVolume* detectorVolume = nullptr;
```

Comment:
Could this have been an `std::variant`?

Reply:
Yes and no (I tried this first), indeed we could have a variant here, but I also need the option that both are `nullptr`, so either it would be a variant with a `std::monostate` to catch the latter case, or a `std::optional<std::variant<...,...>>` which I found overly complicated.

Reply:
Ok. Just seemed like a very custom tagged union to me, but I won't insist on changing anything.